source/common/secret/sds_api.cc

```diff
@@ -133,7 +135,9 @@ void SdsApi::onConfigUpdate(const std::vector<Config::DecodedResourceRef>& resou
 
 void SdsApi::onConfigUpdate(const std::vector<Config::DecodedResourceRef>& added_resources,
                             const Protobuf::RepeatedPtrField<std::string>&, const std::string&) {
-  validateUpdateSize(added_resources.size());
+  if (!validateUpdateSize(added_resources.size())) {
+    return;
+  }
   onConfigUpdate(added_resources, added_resources[0].get().version());
 }
```

Comment:
How is the removal process going to work with this? Seems like we just ignore the update if we get a removal without an addition? 

Is go-control-plane sending two distinct updates, first to remove (which is ignored) then another to add?

Reply:
As per the comments on the issue to follow the RDS handling. For now, I have fixed it as per the RDS handling. 

https://github.com/envoyproxy/envoy/issues/24373

https://github.com/envoyproxy/envoy/blob/5b9183f80d6c7ab38877b8f8f48b152509f49d3e/source/common/rds/rds_route_config_subscription.cc#L120-L134

Reply:
Please let me know if we need to handle it differently. Thanks.

Reply:
It seems to me like both SDS and RDS aren't really supporting xDS removals and just ignore the update in that case, so while this change would probably fix your exception issue it doesn't seem ideal? 

cc @adisuissa here as well, I'm not sure where we are with support for resource removal for the different resource types